src/sync/sync_manager.cpp

```diff
@@ -313,28 +308,37 @@ std::shared_ptr<SyncUser> SyncManager::get_user(const std::string& identity,
     }
 }
 
-std::shared_ptr<SyncUser> SyncManager::get_existing_logged_in_user(const std::string& identity) const
+std::shared_ptr<SyncUser> SyncManager::get_admin_token_user(const std::string& identifier, std::string refresh_token)
 {
     std::lock_guard<std::mutex> lock(m_user_mutex);
-    auto it = m_users.find(identity);
-    if (it == m_users.end()) {
-        return nullptr;
+    auto it = m_admin_token_users.find(identifier);
+    if (it == m_admin_token_users.end()) {
+        // No existing user.
+        auto new_user = std::make_shared<SyncUser>(std::move(refresh_token),
+                                                   identifier,
+                                                   none,
+                                                   SyncUser::TokenType::Admin);
+        m_admin_token_users.insert({ identifier, new_user });
+        return new_user;
+    } else {
+        return it->second;
     }
-    auto ptr = it->second;
-    return (ptr->state() == SyncUser::State::Active ? ptr : nullptr);
 }
 
 std::vector<std::shared_ptr<SyncUser>> SyncManager::all_logged_in_users() const
 {
     std::lock_guard<std::mutex> lock(m_user_mutex);
     std::vector<std::shared_ptr<SyncUser>> users;
-    users.reserve(m_users.size());
+    users.reserve(m_users.size() + m_admin_token_users.size());
     for (auto& it : m_users) {
         auto user = it.second;
         if (user->state() == SyncUser::State::Active) {
             users.emplace_back(std::move(user));
         }
     }
+    for (auto& it : m_admin_token_users) {
+        users.emplace_back(std::move(it.second));
+    }
```

Comment:
Why are we now adding admin token users into this list? Is that in some way related to the general theme of this change?